### DISCUSSION TITLE
Add ability to determine current state of motion and event detection

### DISCUSSION THREAD

**Discussion by ys333193** [blakeblackshear/frigate#13837] [Ideas]
It would be very useful to be able to determine if cameras currently detect motion or events.  

I have a custom camera monitoring applicaiton that plays a slideshow when cameras detect no motion/events, but stitches to show live feed when motion or events are detected.  I use Frigate MQTT events for this, which works great, but on application startup it's impossible to determine current state of motion detection or if camera is detecting events.  Application has to wait until next event, which sometimes isn't triggered for a while.

A simple solution would be for Frigate to publish /camera/motion and/or /camera_activity topic as well as /event topic with Retain flag.  This can be opt-in through configuration to preserve backwards compatibility.

Another option is to expose an HTTP endpoint returning what is currently published under /camera_activity.

Maybe even both options? ;-)

**Comment by hawkeye217** [comment]
There have been issues with the `retain` flag in the past (which is why it was removed), but there's already a feature request to have another look at it again. https://github.com/blakeblackshear/frigate/issues/11314

**Reply by NickM-27** [reply]
it wouldn't make sense to retain these anyway because the state could have changed during the restart

**Reply by ys333193** [reply]
This isn't exactly what I'm looking for.  
I'm not referring to the startup of Frigate itself, but startup of my own app: it starts consuming from MQTT topics and will know when motion/events status has _changed_, but it would not know what it is currnently, and will not know until state changes.
I think it _would_ make sense to retain messages I've mentioned, provided Frigate updates them at startup.  It is true that retail has side effects, that's why I'm suggesting to make it a configurable feature.
Exposing an API endpoint would also have not side effects.

**Reply by NickM-27** [reply]
then you can just send a command with the `onConnect` payload and check the `camera_activity` topic

https://github.com/blakeblackshear/frigate/blob/27e71eb14252fdfa3369b8836412031cef9a7cd6/frigate/comms/dispatcher.py#L143

**Reply by ys333193** [reply]
This would be perfect for my needs, but I can't get it to work: published to frigate/onConnect, but see no frigate/camera_activity message.  Seems like dispatcher.py does not care about payload for onConnect.
I'm using latest frigate:stable docker image.  MQTT is enabled in config, and I only set host and credentials properties.
Just to make sure, I published to frigate/restart and it restarted just fine.

**Reply by NickM-27** [reply]
oh right, the listener is only on websocket not on mqtt, will need to fix that

**Reply by NickM-27** [reply]
will be available in 0.15

**Comment by NickM-27** [comment]
the state of motion detection is already published on the path you specified https://docs.frigate.video/integrations/mqtt#frigatecamera_namemotion

and there are also topics for the count of objects that are published to as soon as objects are detected as well as the `all` topic which encompasses all objects. https://docs.frigate.video/integrations/mqtt#frigatecamera_nameobject_name

**Reply by ys333193** [reply]
Messages published to /fragate/camera/motion only on changes (detection or lack of thereof), but I need to be able to tell what is current state of motion right now.  This would allow me to set the initial values for my app.

**Reply by NickM-27** [reply]
see https://github.com/blakeblackshear/frigate/discussions/13837#discussioncomment-10697084
